subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -357,6 +372,39 @@ int lll_adv_scan_req_report(struct lll_adv *lll, struct pdu_adv *pdu_adv_rx,
 }
 #endif /* CONFIG_BT_CTLR_SCAN_REQ_NOTIFY */
 
+bool lll_adv_connect_ind_check(struct lll_adv *lll, struct pdu_adv *ci,
+			       uint8_t tx_addr, uint8_t *addr,
+			       uint8_t rx_addr, uint8_t *tgt_addr,
+			       uint8_t devmatch_ok, uint8_t *rl_idx)
+{
+	/* LL 4.3.2: filter policy shall be ignored for directed adv */
+	if (tgt_addr) {
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+		return ull_filter_lll_rl_addr_allowed(ci->tx_addr,
+						      ci->connect_ind.init_addr,
+						      rl_idx) &&
+#else
+		return (1) &&
+#endif
+		       isr_rx_ci_adva_check(tx_addr, addr, ci) &&
+		       isr_rx_ci_tgta_check(lll, rx_addr, tgt_addr, ci,
+					    *rl_idx);
+	}
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+	return ((((lll->filter_policy & 0x02) == 0) &&
```

Comment:
Out of scope of this PR, but something I'd like to see in the future: `#define`d values for the filter_policy.